### PR TITLE
Add precompile call data to ContractFunctionResults

### LINKED ISSUE #157: Expand ContractFunctionResult to include FunctionParameters, Amount and GasLimit
### Problem

For precompiled function calls the applicable `FunctionParameters`, `Amount` and `GasLimit` properties stated on call are missing in explicit specification.
This leaves readers of the record stream with missing info or with the problem of being forced to implement heavy logic to recreate the data in an identical manner as the client side.

These values are usually found in the `ContractCreate`/`ContractCall` transaction bodies.
However, in the precompile case the transactions are non contract transactions and are therefore missing as easy properties to read.

### Solution

- Add `FunctionParameters` for the precompile function to the `ContractFunctionResult` proto
- Add `Amount` sent to contract to the `ContractFunctionResult` proto. If not payable set to 0
- Add `GasLimit` to the `ContractFunctionResult` proto. Can match the parent transaction gasLimit

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by shemnon**
**Description**:

Add precompiled call data for parameters, gas, and amount to the
ContractFunctionResults. These would be in the TransactionBody
if it was a normal call but the body for precompiles is a
different type,

Signed-off-by: Danno Ferrin <danno.ferrin@hedera.com>

**Related issue(s)**:

Fixes #157

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: services/contract_call_local.proto
Comment (steven-sheehy) [inline services/contract_call_local.proto:137]: There's already a `gasUsed` that shows the gas used. This would be the gas limit that shows the maximum amount of gas remaining still to use, right?

File: services/contract_call_local.proto
Comment (steven-sheehy) [inline services/contract_call_local.proto:142]: Should we use wrapper types for these fields to clearly indicate not set or is primitive default values sufficient?

File: services/contract_call_local.proto
Comment (SimiHunjan) [reply services/contract_call_local.proto:137]: Is it supposed to represent actual gas used maybe? Related to this issue https://github.com/hashgraph/hedera-services/issues/2949.

Comment (Nana-EC) [reply services/contract_call_local.proto:137]: Steven is right this should represent the `gasLimit`.
`gasUsed` is already captured in item 5 of this proto object

File: services/contract_call_local.proto
Comment (Nana-EC) [inline services/contract_call_local.proto:158]: Is there a max size on the bytes length that's worthy of note here?

**Review by Nana-EC — COMMENTED** [review]
Some suggestions.
Commenting rather than requesting changes so as not to block whiles i'm OOF
Please wait for 1 mirror node team member sign off before merging.

File: services/contract_call_local.proto
Comment (shemnon) [reply services/contract_call_local.proto:142]: There is no meaning for a missing field or a default field here like there is with the tracing valueWritten field, so a wrapper type would just add more bytes to the record.

File: services/contract_call_local.proto
Comment (shemnon) [reply services/contract_call_local.proto:158]: Wrapper type adds no bytes.  The context of the record provides the meaning: is it a child transaction?  Wrapper types are usefull for when null is different from a zero length array.  Nulls won't matter in the code.

Comment (shemnon) [reply services/contract_call_local.proto:158]: No protocol imposed limit.  The functional limit is how much gas you want to pay.

File: services/contract_call_local.proto
Comment (shemnon) [reply services/contract_call_local.proto]: `gas` is the name of the field used in the three paired transaciton types, and I want to try and keep it the same here.  I updated the docs.

Comment (steven-sheehy) [reply services/contract_call_local.proto]: Right, was just arguing for a doc fix here. Looks good

**Review by steven-sheehy — APPROVED** [review]

File: services/contract_call_local.proto
Comment (Nana-EC) [reply services/contract_call_local.proto:158]: Sounds good

**Review by Nana-EC — APPROVED** [review]
LGTM

**Pull request merged**
